### ISSUE TITLE
Difficulty working with repeat softmasked  EDTA output.

### ISSUE THREAD

**Issue by RijanDhakal2055** [NBISweden/AGAT#199]
What I am trying to do: Translate a fasta file accounting for the features detailed in a corresponding GGF3 file.

Perl script that I am using: agat_sp_extract_sequences.pl 

The input files that I am using:  https://drive.google.com/file/d/125IXS4lOhsQRRg79uP4VF5nffcuxn7qp/view?usp=sharing and 
[GFF_file.zip](https://github.com/NBISweden/AGAT/files/7575420/GFF_file.zip)

The flags that I used for AGAT : 
`agat_sp_extract_sequences.pl --gff toomers_hap1.gff3 --fasta toomers_hap1.fasta -t gene --aa -o hap1_aa.fa`

The error: It seems AGAT is unable to accommodate repeat masked files according to this thread [here](https://github.com/NBISweden/AGAT/issues/102). I am using AGAT for the very first time and cannot quite fully understand the fix that you have posted in that issue thread. The main way to solve the problem at is to edit two JSON files? But where are these JSON files? I completed the installation using conda, so will they be in the conda's bin directory?

The exact error message being:
`gff3 reader warning: primary_tag error @ copia_ltr_retrotransposon still not taken in account! Please modify the json files to define the feature in one of the levels.`

Can you help me understand what the issue might be?

**Comment by Juke34** [comment]
The feature type (3rd colun of your file) `copia_ltr_retrotransposon` is unknown by AGAT. 
Run `agat_convert_sp_gxf2gxf.pl --expose` then modify the file called `features_level1.json` to add `"copia_ltr_retrotransposon":"standalone",` as explained here: https://agat.readthedocs.io/en/latest/troubleshooting.html#agat-throws-features-out-because-the-feature-type-is-not-yet-taken-into-account

**Comment by RijanDhakal2055** [comment]
Thank you for your quick response,
I added the `"copia_ltr_retrotransposon":"standalone"` and that solved the issue for "copia_ltr_retrotransposon", I had to gradually add all the missing elements form the third column one by one and these error flags are now gone but the translated file is somehow empty and the final output from agat_sp_extract_sequences.pl says 0 CDS translated, it does mention that 1757 elements were detected.

I found a few threads that mention an empty output from agat_sp_extract_sequences.pl, I think this one seems to imply white spaces in the GFF3 file being the issue: https://github.com/NBISweden/AGAT/issues/179

I am planning to write a python script or a bash script to see if I can get rid of the white-spaces and see if that solves the issue at hand. I tried the "-t exon --merge --aa" flags as you suggested here : https://github.com/NBISweden/AGAT/issues/154 but I am still getting an empty translated fasta file.

I am using the same files that I used in the original comment, except for the modded level1 jsons which is here:
[features_level1.txt](https://github.com/NBISweden/AGAT/files/7591972/features_level1.txt)
 
I have just changed the file extension to .txt because github does not support .json files. I will see if removing whitespaces does the trick for me, otherwise any ideas?
